plugins/filter/lib/filter.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ * This file is a part of the Voice Call Manager project
+ *
+ * Copyright (C) 2024  Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "filter.h"
+#include "filterlist.h"
+
+using namespace VoiceCall::Plugins;
+
+class Filter::Private
+{
+public:
+    Private()
+        : m_rejected(QString::fromLatin1("/sailfish/voicecall/filter/rejected-numbers"))
+        , m_ignored(QString::fromLatin1("/sailfish/voicecall/filter/ignored-numbers"))
+        , m_whitelist(QString::fromLatin1("/sailfish/voicecall/filter/whitelist"))
+    {
+    }
+
+    enum Action {
+          NO_FILTER,
```

Comment:
Just a suggestion, but how about something like FILTER_APPROVE so all the cases would have similar format? Also this can be now returned if there _is_ a filter, i.e. the white list one.

Reply:
Nice suggestion, I'll split the NO_FILTER into a FILTER_APPROVE and a FILTER_NO_MATCH. It makes the various possibilities more straight forward. Of course, both APPROVED and NO_MATCH would let the call pass.